src/main/scala/mesosphere/marathon/core/storage/PersistenceStore.scala

```
@@ -0,0 +1,254 @@
+    */
+  protected def rawIds(parent: K): Source[K, NotUsed]
+
+  /**
+    * Get the data, if any, for the given primary id and value type.
+    *
+    * @return A future representing the data at the given Id, if any exists.
+    *         If there is an underlying storage problem, the future should fail with [[StoreCommandFailedException]]
+    */
+  protected[storage] def rawGet(id: K): Future[Option[Serialized]]
+
+  /**
+    * Set the value at the given id.
+    *
+    * @return A future that completes when the value has been updated to the new value
+    *         or fail with [[StoreCommandFailedException]] if there is an underlying storage issue or the value
+    *         didn't exist.
+    */
+  protected def rawSet(id: K, v: Serialized): Future[Done]
+
+  /**
+    * Create the value at the given id.
+    *
+    * @return A future that completes when the value has been associated with the Id
+    *         or fail with [[StoreCommandFailedException]] if there is an underlying storage issue or the
+    *         value already existed.
+    */
+  protected def rawCreate(id: K, v: Serialized): Future[Done]
+
+  private def createOrUpdate[Id, V](
+    id: Id,
+    change: Option[V] => Try[V])(implicit
+    ir: IdResolver[Id, K, V, Serialized],
+    m: Marshaller[V, Serialized],
+    um: Unmarshaller[Serialized, V],
+    ctx: ExecutionContext = ExecutionContext.global): Future[Option[V]] = async {
+    val path = ir.toStorageId(id)
+    val old = await(get(id))
+    change(old) match {
+      case Success(newValue) =>
+        val serialized = await(Marshal(newValue).to[Serialized])
+        old match {
+          case Some(_) =>
+            await(rawSet(path, serialized))
+            old
+          case None =>
+            await(rawCreate(path, serialized))
+            old
+        }
+      case Failure(error: StoreCommandFailedException) =>
+        throw error
+      case Failure(error) =>
+        old
```

What is the reasoning here to ignore all errors silently?